kairos-cli/bin/commands/mod.rs

```diff
@@ -0,0 +1,15 @@
+pub mod deposit;
+pub mod transfer;
+pub mod withdraw;
+
+use clap::Subcommand;
+
+#[derive(Subcommand)]
+pub enum Command {
+    #[command(about = "Deposits funds into your account")]
+    Deposit(deposit::Args),
```

Comment:
Factoring out these into separate modules increases obscurity. it requires a reader to go to the module and figure out what args the deposit command requires. I'm taking the perspective of someone just wanting to know what our CLI is able to do. I would much more prefer to see that in one file.

I'm also taking into account that you are trying to create a pattern where every command has a `run` function and it solely operates on respective arguments, which I think is great.

But I would suggest creating a `cli` module putting the CLI types in it together with the respective args. This would obviously cause some name collisions for the `Args` type. Args is a very generic name, instead I would suggest introducing `DepositArgs`, `TransferArgs`, and `WithdrawArgs`.

These type names make it very obvious about which arguments we are actually talking and by putting the whole CLI description into one file, someone new to the project only needs to read one module to know what it's capable to do.

Reply:
In case we put CLI 'spec' in one file, then it will become like:

```rs
#[derive(Subcommand)]
pub enum Command {
    #[command(about = "Deposits funds into your account")]
    Deposit {
        #[clap(flatten)]
        amount: AmountArg,
        #[clap(flatten)]
        private_key_path: PrivateKeyPathArg,
    },
    // ...
```

Looks great... but how we can get `DepositArgs` type for deposit handler `pub fn run(args: DepositArgs) -> Result<String, CliError>`?

Maybe some handler-derive macro could work, but I could not find a way to make it work :roll_eyes:.

I thought about different files structure, but merging all those commands into one file does not make sense to me. I would leave it as it is.